### PR TITLE
FPC with Fabric 2.2 and clean(ish) peer build

### LINKED ISSUE #406: Remove need for symbolic links for built-in FPC plugins in Fabric
**Is your feature request related to a problem? Please describe.**
FPC plugins are now compiled together with the peer. However, in order to do so, the Fabric patching procedure has to first create symbolic links within the Fabric tree pointing to FPC plugins folders.

**Describe the solution you'd like**
We want to build the Fabric peer (with the FPC selectable plugins) without having to create such symbolic links.

### PR DESCRIPTION
**Pull request by g2flyer**
**What this PR does / why we need it**:

This makes fpc (including demo with all embellishment like couchdb and explorer) work with fabric v2.2.
Additionally (and necessary as fabric 2.2 contains now a go.mod) it also builds peer and fabric utililities via go-modules.    All symlinks are gone and package imports are what you expect, but go modules themselves are still a bit brittle (see below). So better than before but probably still not upstreamable ...

Notes:
- it requires bi-directional go-modules as modules and vendor make GOPATH ignored.
- we also have to explicitly have to build ignoring vendor as modules plus vendor via the "standard" `go mod vendor` does not vendor all files which lead to cgo problems. 
- currently to make it work i had to rely `replace` clauses pointing to FPC in the GOPATH, also referencing fabric from fpc is also a bit tricky (see comments in go.mod). The latter there is not much we can do about, the former i have to see whether it can be improved once the first version is committed in master. 
- go mods can also be rather brittle (hence maybe Matt's comment that they might not support them) as some slighly newer version of some modules in our go module made everything build nicely but then made configtxgen fail mysteriously at runtime (see also related comment in go.mod)

*PS: Notes on fabric v2.2.0:*
- requires now go 1.14.4
- added new lifecycle command [peer lifecycle chaincode queryapproved](https://hyperledger-fabric.readthedocs.io/en/release-2.2/commands/peerlifecycle.html#peer-lifecycle-chaincode-queryapproved)
- the announcement mail also mentioned [LTS Strategy doc](https://github.com/hyperledger/fabric-rfcs/blob/master/text/0005-lts-release-strategy.md), in case you haven't seen that already ...
- byfn is removed (from 2.1.1+ in fabric-samples)
- see [release notes](https://github.com/hyperledger/fabric/releases/tag/v2.2.0) for more info

**Which issue(s) this PR fixes**:

Next iteration of #370, should fix #406. 

**Special notes for your reviewer**:

Before testing, you will have to upgrade fabric to tag `v2.2.0`.
You also will need to upgrade go to 1.14.4.

**Does this PR introduce a user-facing changes and/or breaks backward compatability?**:

```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -2,37 +2,34 @@ module github.com/hyperledger-labs/fabric-private-chaincode
...
-	github.com/fsouza/go-dockerclient v1.6.5 // indirect
 	github.com/gin-contrib/cors v1.3.1
 	github.com/gin-gonic/gin v1.6.1
 	github.com/golang/protobuf v1.3.5
-	github.com/hashicorp/go-version v1.2.0 // indirect
-	github.com/hyperledger-labs/fabric-private-chaincode/ercc v0.0.0-20200323171133-70f103dd66ee
-	github.com/hyperledger/fabric v2.1.1+incompatible
-	github.com/hyperledger/fabric-amcl v0.0.0-20200424173818-327c9e2cf77a // indirect
+	github.com/hyperledger-labs/fabric-private-chaincode/ercc v0.0.0-20200708113644-adeb2609f5e7
+	github.com/hyperledger/fabric v1.4.0-rc1.0.20200715015833-3741860ac90f
```
Comment (bvavala) [inline go.mod:26]: v1.4.0?

File: integration/config/core.yaml [hunk 2]
```diff
@@ -161,6 +165,10 @@ peer:
         aliveExpirationTimeout: 25s
         # Reconnect interval(unit: second)
         reconnectInterval: 25s
+        # Max number of attempts to connect to a peer
+        maxConnectionAttempts: 120
```
Comment (bvavala) [inline integration/config/core.yaml:169]: if things do not work, we would have to wait 120 attempts \times 25 seconds = 50min for the peer(?) to fail.
I guess 2-3 should be enough.

Comment (g2flyer) [reply integration/config/core.yaml:169]: this is the default from sampleconfig. Unless we have a good reason to change it, i would leave default configs as-is ...

File: go.mod (hunk 1 above)
Comment (g2flyer) [reply go.mod:26]: so comment on top of file ....  (fabric api versioning is not consistent with go module conventions and so you cannot use v2.2.0, but the comment corresponds to the same)

File: integration/config/core.yaml [hunk 3]
```diff
@@ -28,23 +28,26 @@ peer:
     # The endpoint this peer uses to listen for inbound chaincode connections.
     # If this is commented-out, the listen address is selected to be
     # the peer's address (see below) with port 7052
-    chaincodeListenAddress: 0.0.0.0:7052 # Note: we uncommented it to make working with our dev docker image
+    chaincodeListenAddress: 0.0.0.0:7052 # FPC change: we uncommented it to make working with our dev docker image
```
[the remark this answers is not in the document]
Comment (g2flyer) [reply integration/config/core.yaml:31]: i think it was related to the other address related changes, some of them related to proxies.  Nothing changed in this respect related to fabric v2.2, i just changed the comment that it was fpc related ...

File: utils/docker-compose/network-config/configtx.yaml [hunk 4]
```diff
@@ -3,7 +3,7 @@
 #
 # SPDX-License-Identifier: Apache-2.0
 #
-# Based on first-network/configtx.yaml from github.com/hyperledger/fabric-samples.git tagged with v2.1.1
+# Based on first-network/configtx.yaml from github.com/hyperledger/fabric-samples.git tagged with v2.1.0
```
[the remark this answers is not in the document]
Comment (g2flyer) [reply utils/docker-compose/network-config/configtx.yaml:6]: nope, there is (at least was not as of thursday) any 2.2 of fabric samples. more importantly, first-network is anyway going away (was removed in 2.1.1.).  I was looking in test-network, the new "standard",  whether i should rebase it on that, but didn't seem worth at this point in time. Maybe as part of any refactoring going to multi-org could change our docker-compose stuff to be closer to test-network. Would be probably good in the long run but not sure whether worth the effort in short term ....

File: integration/config/core.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (mbrandenburger) [reply integration/config/core.yaml:31]: I have the feeling this comment can be removed

File: utils/docker-compose/network-config/configtx.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (mbrandenburger) [reply utils/docker-compose/network-config/configtx.yaml:6]: aha, ok - I though this was just a typo.

File: integration/config/core.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (g2flyer) [reply integration/config/core.yaml:31]: hmm, did you test that it still works in all circumstances when you remove the comment _and_ comment out the definition (as in the original fabric version)?

[the remark this answers is not in the document]
Comment (mbrandenburger) [reply integration/config/core.yaml:31]: ok, I just got confused, in the original sampleconfig/core.yaml this line is commented out .... sorry ... I was wrong. all fine

**Pull request merged**